common/thanos/templates/alerts/thanos-alerts.yaml

```diff
@@ -16,19 +16,39 @@ metadata:
 spec:
 {{ include $path (list $name $root) | indent 2 }}
 
+{{- end }}
+{{- else if and $.Values.ruler.enabled (eq $.Values.deployWholeThanos false) }}
+{{- range $component := tuple "query" "ruler" }}
```

Comment:
it's a little bit unfortunate, that this file has the same code three times. It reads ok but it takes a while to digest. Maybe we can add a comment in the beginning which would help my further self to faster parse this file.

> Alerts are deployed dependent upon the components chosen.  It could either be `Compactor|Query|Store`, `Query|Ruler` or `Query`.